.github/workflows/deploy-draft.yml

```diff
@@ -0,0 +1,86 @@
+name: Deploy branch draft on Netlify
+on:
+  pull_request:
+    types: [labeled, opened, synchronize]
+    
+jobs:
+  build-deploy:
+    if: |
+      (github.event.action == 'labeled' && github.event.label.name == 'deploy') ||
+      (github.event.action != 'labeled' && contains(github.event.pull_request.labels.*.name, 'deploy'))
+    runs-on: ubuntu-latest
+    steps:
+    - name: inject slug/short variables
+      uses: rlespinasse/github-slug-action@v3.x
+
+    - name: set STAGE variable in environment for next steps
+      run: echo "STAGE=pr-${{ github.event.number }}-${{ env.GITHUB_HEAD_REF_SLUG }}" >> $GITHUB_ENV
+
+    - name: create a github deployment
+      uses: bobheadxi/deployments@v0.5.2
+      id: deployment
+      with:
+        step: start
+        token: ${{ secrets.GITHUB_TOKEN }}
+        env: ${{ env.STAGE }}
+        ref: ${{ github.head_ref }}
+        
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0 # fetch whole repo so git-restore-mtime can work
+
+    - name: Adjust file watchers limit
+      run: echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
+
+    - uses: actions/setup-node@v1
+      with:
+        node-version: '14.x'
+    - name: Install yarn
+      run: sudo npm -g install yarn
+    - name: Yarn install
+      run: yarn install --immutable
+      env:
+        NODE_ENV: ${{ secrets.NODE_ENV }}
+
+    - name: Checking Gatsby cache
+      id: gatsby-cache-build
+      uses: actions/cache@v2
+      with:
+        path: |
+          public
+          .cache
+        key: ${{ runner.os }}-gatsby-build-develop-${{ github.run_id }}
+        restore-keys: |
+          ${{ runner.os }}-gatsby-build-develop-
+
+    - name: Fix mtimes
+      run: yarn fix-mtimes --force
+    - name: Gatsby build
+      run: yarn build
+      env:
+        APP_ENV: staging
+        NODE_ENV: ${{ secrets.NODE_ENV }}
+        NODE_OPTIONS: --max-old-space-size=4096
+        ALGOLIA_API_KEY: ${{ secrets.ALGOLIA_API_KEY }}
+        ALGOLIA_APP_ID: ${{ secrets.ALGOLIA_APP_ID }}
+        ALGOLIA_INDEX_NAME: edb-docs-staging
+        INDEX_ON_BUILD: false
+
+    - name: Netlify deploy
+      run: |
+        sudo yarn global add netlify-cli
+        netlify deploy --dir=public --json | tee deploy_results.json
+        echo "NETLIFY_DRAFT_URL=`jq -r '.deploy_url' deploy_results.json`" >> $GITHUB_ENV
```

Comment:
Same as above, we could add `netlify-cli` as a dev dependency to the package.json or use this other action? https://github.com/marketplace/actions/netlify-actions